protocol/x/clob/keeper/process_single_match.go

```diff
@@ -271,29 +271,45 @@ func (k Keeper) persistMatchedOrders(
 	}
 
 	if !isTakerLiquidation {
-		if matchWithOrders.TakerOrder.MustGetOrder().RouterSubaccountId != nil && bigRouterTakerFeeQuoteQuantums.Sign() != 0 {
-			fmt.Printf("bigRouterTakerFeeQuoteQuantums: %v\n", bigRouterTakerFeeQuoteQuantums)
-			updates = append(updates, satypes.Update{
-				AssetUpdates: []satypes.AssetUpdate{
-					{
-						AssetId:          assettypes.AssetTDai.Id,
-						BigQuantumsDelta: bigRouterTakerFeeQuoteQuantums,
+		takerRouter := matchWithOrders.TakerOrder.MustGetOrder().RouterSubaccountId
+		makerRouter := matchWithOrders.MakerOrder.MustGetOrder().RouterSubaccountId
+
+		if takerRouter != nil && makerRouter != nil && *takerRouter == *makerRouter {
+			totalRouterFee := new(big.Int).Add(bigRouterTakerFeeQuoteQuantums, bigRouterMakerFeeQuoteQuantums)
+			if totalRouterFee.Sign() != 0 {
```

Comment:
perhaps we panic here if totalRouterFee is somehow negative, or even better if either maker or taker router fee is negative